### PR TITLE
Get started with PP stdlib specs, especially the PP::pp method

### PR DESCRIPTION
**Pull request by wied03**

### CODE REVIEW

File: library/pp/mcall_spec.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+require 'pp'
+
+describe "PP.mcall" do
```
Comment (eregon) [inline library/pp/mcall_spec.rb]: PP.mcall seems to be intended as private, so should not be spec'd, can you remove the file?

File: library/pp/objectmixin/pretty_print_cycle_spec.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+require File.expand_path('../../../../spec_helper', __FILE__)
+require 'pp'
+
+describe "PP::ObjectMixin#pretty_print_cycle" do
```
Comment (eregon) [inline library/pp/objectmixin/pretty_print_cycle_spec.rb]: The PP::ObjectMixin module is an implementation detail, can you remove these files?

File: library/pp/pp_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    $stdout = @original_stdout
+  end
+  
+  it 'works with default arguments' do
+    $stdout = StringIO.new    
+    array = [1, 2, 3]
+    
+    PP.pp array
+    
+    $stdout.string.should == "[1, 2, 3]\n"
```
Comment (eregon) [inline library/pp/pp_spec.rb]: There is a nice matcher for output :wink: 
```ruby
lambda {
  PP.pp array
}.should output("[1, 2, 3]\n")
```

File: library/pp/pp_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    
+  it 'allows specifying out explicitly' do
+    $stdout = StringIO.new
+    other_out = StringIO.new
+    array = [1, 2, 3]
+    
+    PP.pp array, other_out
+    
+    other_out.string.should == "[1, 2, 3]\n"
+    $stdout.string.should == ''
```
Comment (eregon) [inline library/pp/pp_spec.rb]: Try to write this one with the `output` matcher as well (`should output("")`).
The `new_io` helper should be handy for testing the other output without requiring StringIO.

File: library/pp/pp_spec.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  end
+  
+  it "needs to be reviewed for spec completeness"
```
Comment (eregon) [inline library/pp/pp_spec.rb]: This should be removed as we now have specs for this method :smile:

**Comment by eregon** [comment]
Looks good!
Could you reduce the PR to just the `library/pp/pp_spec.rb` file? It seems PP does not use private methods very well and the automatic generator does not seem so useful in this case.

File: library/pp/mcall_spec.rb (hunk 1 above)
Comment (wied03) [inline library/pp/mcall_spec.rb]: Sure. That was the generator. I saw some other specs that had the skeletons still there, so I followed that, but will be happy to remove these.

File: library/pp/pp_spec.rb (hunk 5 above)
Comment (wied03) [inline library/pp/pp_spec.rb]: Well I didn't test every aspect of this method, just wanted to get the ball rolling. Do you still want to remove that?

**Comment by wied03** [comment]
@eregon - I used the output matcher for the 1st spec. The only problem for the other is that the main reason I'm doing this is to improve Opal's compatibility with MRI. Opal doesn't support File I/O and new_io, output_with_fd, etc. all need temporary files in order to work.

File: library/pp/pp_spec.rb (hunk 5 above)
Comment (eregon) [inline library/pp/pp_spec.rb]: Right, it's fine to leave as is then.

**Comment by eregon** [comment]
I see, would using IOStub (defined in mspec) work?
It's a simpler StringIO and it means we don't rely on the impl having available StringIO.

**Comment by wied03** [comment]
@eregon - No problem, I think IOSub will work just fine, pushed a new commit with that

**Comment by eregon** [comment]
@wied03 Great, thanks and merged!

**Comment by eregon** [comment]
Thank you for you contribution!

**Pull request merged**